tests/sentry/integrations/jira_server/test_integration.py

```diff
@@ -935,6 +913,119 @@ def test_sync_assignee_outbound_api_error(self, mock_assign_issue):
+            external_issue_key="APP-123",
+            user=rpc_user,
+            logging_context={"integration_id": self.integration.id},
+        )
+
+        mock_search_users_for_issue.return_value = [
+            {
+                "name": "jirauser",
+                "displayName": "Jira User",
+                "emailAddress": "some_email@example.com",
+            }
+        ]
+
+        assert result is None
+
+    @responses.activate
+    @patch("sentry.integrations.jira_server.integration.JiraServerClient.search_users_for_issue")
+    def test_get_user_by_external_actor_no_matching_jira_user(self, mock_search_users_for_issue):
+        rpc_user = serialize_rpc_user(self.create_user(email="bob@example.com"))
+        ExternalActor.objects.create(
+            organization_id=self.organization.id,
+            integration_id=self.integration.id,
+            provider=ExternalProviders.JIRA_SERVER.value,
+            user_id=rpc_user.id,
+            external_name="nonexistent_user",
+        )
+
+        mock_search_users_for_issue.return_value = []
+
+        result = self.installation._get_user_by_external_actor(
+            client=self.installation.get_client(),
+            external_issue_key="APP-123",
+            user=rpc_user,
+            logging_context={"integration_id": self.integration.id},
+        )
+
+        assert result is None
+        mock_search_users_for_issue.assert_called_once_with("APP-123", "nonexistent_user")
+
+    @responses.activate
+    @patch("sentry.integrations.jira_server.integration.JiraServerClient.search_users_for_issue")
+    def test_get_user_by_external_actor_matching_user(self, mock_search_users_for_issue):
+        rpc_user = serialize_rpc_user(self.create_user(email="bob@example.com"))
+        ExternalActor.objects.create(
+            organization_id=self.organization.id,
+            integration_id=self.integration.id,
+            provider=ExternalProviders.JIRA_SERVER.value,
+            user_id=rpc_user.id,
+            external_name="JiraUser",
+        )
+
+        mock_jira_user = {
+            "name": "jirauser",
+            "displayName": "Jira User",
+            "emailAddress": "some_other_email@example.com",
+        }
+
+        # Tests case where a username query matches multiple users,
+        # we expect an exact match on the name
+        mock_search_users_for_issue.return_value = [
+            {"name": "other_user", "displayName": "Other User"},
+            mock_jira_user,
+            {"name": "another_user", "displayName": "Another User"},
+        ]
+
+        result = self.installation._get_user_by_external_actor(
+            client=self.installation.get_client(),
+            external_issue_key="APP-123",
+            user=rpc_user,
+            logging_context={"integration_id": self.integration.id},
+        )
+
+        assert result == mock_jira_user
+        mock_search_users_for_issue.assert_called_once_with("APP-123", "JiraUser")
+
```

Comment:
Could add a test for when we default back to search via email when we cant find anyone via username. But Idt it's too important 🤷‍♀️

Reply:
The current tests exercise this logic already so that should be covered.